Marlin/ultralcd.cpp

```diff
@@ -805,13 +807,29 @@ static void lcd_prepare_menu() {
 
 #endif // DELTA_CALIBRATION_MENU
 
-inline void line_to_current(AxisEnum axis) {
-  #if ENABLED(DELTA)
-    calculate_delta(current_position);
-    plan_buffer_line(delta[X_AXIS], delta[Y_AXIS], delta[Z_AXIS], current_position[E_AXIS], manual_feedrate[axis]/60, active_extruder);
-  #else
-    plan_buffer_line(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS], current_position[E_AXIS], manual_feedrate[axis]/60, active_extruder);
-  #endif
+/**
+ * If the manual move hasn't been fed to the planner yet,
+ * and the planner can accept one, send immediately
+ */
+inline void manage_manual_move() {
+  if (manual_move_axis != (int8_t)NO_AXIS && !planner_is_full()) {
+    #if ENABLED(DELTA)
```

Comment:
I don't see how any 16-move wait could occur. Both `movesplanned() < 3` and `planner_is_full()` have the same "problem." Both of them, if they are failing, must wait until some previously-added move finishes. In the first case, it will be a move that was added a couple moves ago. In the latter case, it will be the move that was added perhaps several moves ago, or perhaps just a couple of moves ago. In both cases, if the check fails, and then a move is added before we come back to our check in the next loop (not likely, since we're probably not actively printing), it will continue to fail until finally there is a single slot (or more) available in the buffer. The only difference I see is that `movesplanned() < 3` will begin skipping moves sooner than `planner_is_full`. The moves it ends up skipping are not guaranteed to be shorter, necessarily, since the encoder values still keep going up.

Anyway, we can probably test this scientifically and see what actually happens. I would be curious to know.

In any case, the current solution seems pretty reasonable. I was thinking of the idea of having the moves be delayed until the encoder stops for 1/4 second. But now I think it's not a good idea. It will tend to encourage much longer single moves, which would especially be a problem with the E axis. I think it's better if either (A) the movement happens quickly as a reaction to turning the knob, or (B) the movement waits until the knob is clicked.